src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3297,20 +3299,43 @@ protected override IPointerTypeSymbol CommonCreatePointerTypeSymbol(ITypeSymbol
                 throw new ArgumentException(CSharpResources.OutIsNotValidForReturn);
             }
 
+            if (callingConvention != SignatureCallingConvention.Unmanaged && !callingConventionTypes.IsDefaultOrEmpty)
+            {
+                throw new ArgumentException(string.Format(CSharpResources.CallingConventionTypesRequireUnmanaged, nameof(callingConventionTypes), nameof(callingConvention)));
+            }
+
+            if (!callingConvention.IsValid() || callingConvention == SignatureCallingConvention.VarArgs)
```

Comment:
Is there any risk of a the compiler ever giving us a symbol with SignatureCallingConvention.VarArgs, even if it's unsupported? Because then there's a risk that we hand function pointer symbol to this and things go boom.

Reply:
Possibly, but I think I should change the metadata reading to just throw if it encounters a varargs signature, at which point you can never observe a function pointer with that.

---
In reply to: [463842007](https://github.com/dotnet/roslyn/pull/46467#discussion_r463842007) [](ancestors = 463842007)

Reply:
Actually, on second thought, I should just accept `varargs` here. It should be fine. You can't actually use it (you get a use-site diagnostic), but it's possible to observe from metadata. There's no sane syntax to generate, but you can just treat it like managed and errors will be decent enough.

---
In reply to: [463862706](https://github.com/dotnet/roslyn/pull/46467#discussion_r463862706) [](ancestors = 463862706,463842007)